From 392afe98c9fd000023b4d79e4ff0541e625b4da8 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:23:27 -0400
Subject: [cpufreq] stats: Fix error handling in __cpufreq_stats_create_table()

Message-id: <1409235879-1283-114-git-send-email-prarit@redhat.com>
Patchwork-id: 89478
O-Subject: [RHEL7.1 PATCH BZ 1134639 113/185] cpufreq: stats: Fix error handling in __cpufreq_stats_create_table()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 0b7528d963e7dda51dd07fa7b1f14a9bf03b1cee
Author: Saravana Kannan <skannan@codeaurora.org>
Date:   Thu Feb 27 17:58:35 2014 -0800

    cpufreq: stats: Fix error handling in __cpufreq_stats_create_table()

    Remove sysfs group if __cpufreq_stats_create_table() fails after creating
    one.

    Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Saravana Kannan <skannan@codeaurora.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq_stats.c b/drivers/cpufreq/cpufreq_stats.c
index 22e79c9..568f1d5 100644
--- a/drivers/cpufreq/cpufreq_stats.c
+++ b/drivers/cpufreq/cpufreq_stats.c
@@ -224,7 +224,7 @@ static int __cpufreq_stats_create_table(struct cpufreq_policy *policy,
  stat->time_in_state = kzalloc(alloc_size, GFP_KERNEL);
  if (!stat->time_in_state) {
   ret = -ENOMEM;
-  goto error_out;
+  goto error_alloc;
  }
  stat->freq_table = (unsigned int *)(stat->time_in_state + count);
 
@@ -246,6 +246,8 @@ static int __cpufreq_stats_create_table(struct cpufreq_policy *policy,
  spin_unlock(&cpufreq_stats_lock);
  cpufreq_cpu_put(current_policy);
  return 0;
+error_alloc:
+ sysfs_remove_group(&policy->kobj, &stats_attr_group);
 error_out:
  cpufreq_cpu_put(current_policy);
 error_get_fail:
-- 
1.7.1